src/frontend/python/chiquito/dsl.py

```diff
@@ -99,7 +168,11 @@ def pragma_disable_q_enable(self: Circuit) -> None:
 
     def new_table(self: Circuit, table: LookupTable) -> LookupTable:
         assert self.mode == CircuitMode.SETUP
-        self.tables.add(table)
+        if self.super_circuit is None:
+            raise SyntaxError(
+                "Circuit: new_table() is only available for Circuit with initiated super_circuit field."
+            )
+        self.super_circuit.tables[table.uuid] = table
```

Comment:
Note that here I opted to store the tables under supercircuit and then circuit/steptype both have reference to the supercircuit, so they can reference the tables as well.